graph/collection-properties.md

```diff
@@ -0,0 +1,402 @@
+> Collection properties...provided in the payload corresponding to updatable properties MUST replace the value of the corresponding property in the entity or complex type.
+
+The standard also specifies that collections of entities can be updated in a relative fashion using the [delta syntax](https://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#_Toc38457777) in a `PATCH` request:
+> The body of a PATCH request to a URL identifying a collection of entities...MUST contain the context control information...and...MUST contain an array-valued property named value containing all added, changed, or deleted entities...
+
+### Complex Type
+
+#### Replace the elements in a collection
+
+```HTTP
+PATCH /interestingData
+{
+  "foos": [
+    {
+      "someProperty": "a replacement value"
+    },
+    {
+      "someProperty": "more replacement value"
+    },
+    {
+      "someProperty": "a new value demonstrating that additional elements can be in the collection"
+    }
+  ]
+}
+
+204 No Content
+```
+
+#### Check the new contents of the collection
+
+```HTTP
+GET /interestingData/foos
+
+200 OK
+{
+  "value": [
+    {
+      "someProperty": "a replacement value"
+    },
+    {
+      "someProperty": "more replacement value"
+    },
+    {
+      "someProperty": "a new value demonstrating that additional elements can be in the collection"
+    }
+  ]
+}
+```
+
+### Entity Type
+
+#### Replace the elements in a collection
+
+```HTTP
+PATCH /interestingData
+{
+  "bars": [
+    {
+      "id": "fourthBarId",
+      "differentProperty": 20
+    },
+    {
+      "id": "fifthBarId",
+      "differentProperty": -99999
+    }
+  ]
+}
+
+204 No Content
+```
+
+#### Check the new contents of the collection
+
+```HTTP
+GET /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "fourthBarId",
+      "differentProperty": 20
+    },
+    {
+      "id": "fifthBarId",
+      "differentProperty": -99999
+    }
+  ]
+}
+```
+
+#### Change the elements in a collection
+
+```HTTP
+PATCH /interestingData/bars
+{
+  "@context": "#$delta",
+  "value": [
+    {
+      "id": "fifthBarId",
+      "differentProperty": 1024
+    },
+    {
+      "id": "sixthBarId",
+      "differentProperty": 9801
+    },
+    {
+      "id": "fourthBarId",
+      "@removed": {}
+    }
+  ]
+}
+
+204 No Content
+```
+
+#### Check the new contents of the collection
+
+```HTTP
+GET /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "fifthBarId",
+      "differentProperty": 1024
+    },
+    {
+      "id": "sixthBarId",
+      "differentProperty": 9801
+    }
+  ]
+}
+```
+
+## Exceptions
+
+It might not make sense for a collection property to be a collection of entities in cases where the collection must be viewed holistically; that is, cases where one element in the collection impacts or is coupled with another element in the collection. 
```

Comment:
OData does define the concept of [ordered collections](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_PositionalInserts), with the ability to insert within the ordered collection according to ordinal. We could choose to adopt that pattern in graph, which would help facilitate these types of scenarios on graph and not fall back to a collection of complex types.